### PR TITLE
temp workaround for false approximate match

### PR DESCRIPTION
**Pull request by swanderz**
fixes: #11

#8 _might_ be closer to the permanent solution, but I'm still bottoming out on root cause and will likely need to meet w/ @jtcohen6 sometime next week.

this isn't a permanent fix, but it will work in the short-term, while we dive deep on the source of the `adapter.get_relation` issue

an overly-simplified normal operation for the table materialization is:
1. get the database, schema and identifier of the relation we are trying to make
2. go look and see if a relation with the same three-part name already exists and assign result to `old_relation`
    1. if yes, return an `Relation` object with the three-part name, and a `type` property (so we know if it exists as a `view` or as a `table`)
    2. if no, return `None`
3. if `old_relation` is not `None`, drop the pre-existing relation.

however, because we currently do not support views, if we assume that there are no views currently in the database, then we don't need to step 2 above. instead we can always just call `DROP TABLE <TABLE_NAME> IF EXISTS`, because:
1. the relation should always exist, and
2. if the relation doesn't already exist, the statement won't fail because of the `... IF EXISTS` clause

### CODE REVIEW

**Comment by octavianzarzu30** [comment]
verified it works.

**Review by octavianzarzu30 — APPROVED** [review]

**Comment by swanderz** [comment]
@octavianzarzu30 if this looks good to you, I can release this change as 0.21.3 tomorrow.

**Comment by octavianzarzu30** [comment]
yep looks good, have even tried it today :)

File: dbt/include/firebolt/macros/materializations/table.sql [hunk 1]
```diff
@@ -2,21 +2,19 @@
...
   {%- set target_relation = api.Relation.create(database=database,
                                                 schema=schema,
                                                 identifier=identifier,
                                                 type='table') -%}
-  {%- set exists_as_table = (old_relation is not none and old_relation.is_table) -%}
-  {%- set exists_as_view = (old_relation is not none and old_relation.is_view) -%}
+  {%- set target_relation_view = api.Relation.create(database=database,
+                                                schema=schema,
+                                                identifier=identifier,
+                                                type='view') -%}
```
Comment (ima-hima) [inline dbt/include/firebolt/macros/materializations/table.sql:12]:
```suggestion
  {%- set target_relation_view = api.Relation.create(database=database,
                                                    schema=schema,
                                                    identifier=identifier,
                                                    type='view') -%}
```

Comment (ima-hima) [reply dbt/include/firebolt/macros/materializations/table.sql:12]: Iirc, per PEP8, to have a four-space dedent none of the arguments can appear on the first line. I.e. `database` would need to move to the next line.

Mostly, though, it just didn't look good _to me_. 😉

File: dbt/include/firebolt/macros/materializations/table.sql [hunk 2]
```diff
@@ -2,21 +2,19 @@
 {# Note that a nearly identical materialization appears
    in table.sql #}
   {%- set identifier = model['alias'] -%} {# alias comes from where? #}
-  {%- set old_relation = adapter.get_relation(database=database,
-                                              schema=schema,
-                                              identifier=identifier) -%}
   {%- set target_relation = api.Relation.create(database=database,
                                                 schema=schema,
                                                 identifier=identifier,
                                                 type='table') -%}
```
Comment (ima-hima) [inline dbt/include/firebolt/macros/materializations/table.sql:8]: Can you write some comments somewhere in this code to explain what it's doing? I'm really not clear, so I'm sure a more naive reader would have even more problems.

**Review by ima-hima — CHANGES_REQUESTED** [review]
One formatting change, one documentation request.

File: dbt/include/firebolt/macros/materializations/table.sql (hunk 2 above)
Comment (swanderz) [reply dbt/include/firebolt/macros/materializations/table.sql:8]: Happy to flesh out a comment but this code isn't going to live long. As a start, is the description given in this PR helpful?

Comment (ima-hima) [reply dbt/include/firebolt/macros/materializations/table.sql:8]: Yeah, that's helpful. I think I missed it when I clicked on the link in the email I received. Maybe when the page loaded it scrolled past the PR comment. Or maybe I was so eager to get to the code that I missed it, in which case, just chalk it up to me being me, I guess. 😞 

Out of curiosity, what's actually causing the problem?

Comment (swanderz) [reply dbt/include/firebolt/macros/materializations/table.sql:8]: check out https://github.com/dbt-labs/dbt-core/issues/4187. it seems there's a weird approximate match failure happening that'll take another week to iron out

Comment (swanderz) [reply dbt/include/firebolt/macros/materializations/table.sql]: Just added some context inline with jinja comment string.

**Review by ima-hima — APPROVED** [review]
lgtm. 

And thanks for the additional comment. It actually did help my conception.

**Pull request merged**